### PR TITLE
Implemented #298 feature view history

### PR DESCRIPTION
**Pull request by viktor-podzigun**
In this PR feature view history #298 was implemented.

### CODE REVIEW

File: src/ncwin.cpp [hunk 1]
```diff
@@ -2661,11 +2689,12 @@ void NCWin::EditExit()
 
 	clPtr<FS> fs = _editor.GetFS();
 
-	if ( !fs.IsNull() && g_WcmConfig.editSavePos )
```
Comment (corporateshark) [inline src/ncwin.cpp]: Why is the config flag gone?

Comment (viktor-podzigun) [inline src/ncwin.cpp]: This check was moved inside new methods that work with edit/view history.

Comment (corporateshark) [inline src/ncwin.cpp]: Ok

**Comment by viktor-podzigun** [comment]
Build FAILED.
"C:\projects\walcommander\wcm_2013.sln" (default target) (1) ->
(ValidateSolutionConfiguration target) ->
C:\projects\walcommander\wcm_2013.sln.metaproj : error MSB4126: The specified solution configuration "Debug|Win32|Win32" is invalid. Please specify a valid solution configuration using the Configuration and Platform properties (e.g. MSBuild.exe Solution.sln /p:Configuration=Debug /p:Platform="Any CPU") or leave those properties blank to use the default solution configuration. [C:\projects\walcommander\wcm_2013.sln]

Looks like the same build error in current master.
Could it be fixed in the master, so I can rebase this PR?

**Comment by corporateshark** [comment]
Sure

**Comment by helgp** [comment]
My findings. You may decide to expand some of them to CR. 

- 'History' registry entries - better to move to 'History' registry subfolder. At least, name the entries with 'History' prefix, not the suffix. In general: most common name comes first
- Do not pollute app namespace
LoadViewHistory() -> static History::ViewLoad() et.al.
- class PathList: use proper STL class instead. We use STL instead of writing our own container classes.

**Comment by viktor-podzigun** [comment]
>> Regarding 'History' entries
Agreed (I also thought about it) may be at some point we will have refactoring of configuration, so can do it then. I personally don't like that we store config in registry for Windows and in files for other systems.;

>> Do not pollute app namespace 
Agreed, will try to use static class methods instead global functions where appropriate;

>> class PathList: use proper STL class instead 
Agreed, but in this concrete case its a think wrapper that contains common functionality;

Thanks @helgp for your feedback.

**Comment by corporateshark** [comment]
@viktor-podzigun Please, try to use STL in the new code whenever possible.

**Pull request merged**